code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -516,61 +563,70 @@ GLOBAL_VAR(medibot_unique_id_gen)
 			soft_reset()
 			break
 
-		var/treatment_method = null
 
-		if(C.getBruteLoss() >= heal_threshold)
-			treatment_method = BRUTE
+		var/treat_behaviour // what this medibot will do?
 
-		else if(C.getFireLoss() >= heal_threshold)
-			treatment_method = BURN
+		if(reagent_glass?.reagents.total_volume) //have a beaker with something?
+			treat_behaviour = MEDIBOT_TREAT_INJECT
 
-		else if(C.getOxyLoss() >= (5 + heal_threshold))
-			treatment_method = OXY
+			for(var/datum/reagent/each_beaker_reagent in reagent_glass.reagents.reagent_list)
+				if(!C)
+					continue
```

Comment:
redundant check. `!C` means a patient doesn't exist, but we are here because they exist.